pkg/webhooks/validation/controller/controller.go

```diff
@@ -330,8 +333,7 @@ func (c *Controller) readyForFailClose() bool {
 	if !c.dryRunOfInvalidConfigRejected {
 		if rejected, reason := c.isDryRunOfInvalidConfigRejected(); !rejected {
 			scope.Infof("Not ready to switch validation to fail-closed: %v", reason)
-			req := &reconcileRequest{"retry dry-run creation of invalid config"}
-			c.queue.AddAfter(req, time.Second)
+			c.queue.AddRateLimited(dryRunCreationReq)
```

Comment:
I am not sure this works, since we later do `c.queue.Forget(obj)` which resets the rate limiter? I think instead of adding to the queue here we should return an error up to `processNextWorkItem` from `reconcileRequest` and have a single place to add/forget the queue

Reply:
is fail-open a valid configuration for the webhook?  propagating the error up would change that semantic.

Reply:
i've added a retry bool as a hack around it. what do you think?